### PR TITLE
docs: Document migration to the new Python handler

### PR DESCRIPTION
**Pull request by pawamoy**

### CODE REVIEW

File: docs/handlers/overview.md [hunk 1]
```diff
@@ -4,8 +4,169 @@ A handler is what makes it possible to collect and render documentation for a pa
...
+```toml
+# PEP 621 dependencies declaration
+# adapt to your dependencies manager
+[project]
+dependencies = [
+    "mkdocstrings[python-legacy]>=0.18",
+]
+```
+
+The legacy handler will keep "working" for many releases,
```
Comment (pawamoy) [inline docs/handlers/overview.md:36]:
```suggestion
The legacy handler will continue to "work" for many releases,
```

File: docs/handlers/overview.md [hunk 2]
```diff
@@ -4,8 +4,160 @@ A handler is what makes it possible to collect and render documentation for a pa
...
+
+To use the new, experimental Python handler,
+you must specify the `python` extra when depending on *mkdocstrings*:
+
+```toml
+# PEP 621 dependencies declaration
+# adapt to your dependencies manager
+[project]
+dependencies = [
+    "mkdocstrings[python]>=0.18",
```
Comment (oprypin) [inline docs/handlers/overview.md:61]: As said elsewhere, why not just suggest depending on `mkdocstrings-python`?
I think people are more likely to need to choose a version of `mkdocstrings-python` and ensure that the version of *that* is fresh enough, and then `mkdocstrings` version will be ensured fresh enough as a dependency of `mkdocstrings-python`.

Comment (pawamoy) [reply docs/handlers/overview.md:61]: And as answered there, I like the idea of promoting some handlers as the officially supported ones.
The opposite is true: using the extra would ensure the handler's version is fresh enough and up to date with mkdocstrings' version.

I don't know. Maybe extras are indeed too far fetched. I just think there nice to have.

File: docs/handlers/overview.md [hunk 3]
```diff
@@ -4,8 +4,160 @@ A handler is what makes it possible to collect and render documentation for a pa
...
+[`PYTHONWARNINGS` environment variable](https://docs.python.org/3/library/warnings.html#describing-warning-filters):
+
+```bash
+PYTHONWARNINGS=ignore::UserWarning:mkdocstrings.handlers.python
+```
+
+### Migrate to the experimental Python handler
+
+To use the new, experimental Python handler,
+you must specify the `python` extra when depending on *mkdocstrings*:
```
Comment (pawamoy) [inline docs/handlers/overview.md:54]:
```suggestion
you can depend on `mkdocstrings-python` directly,
or specify the `python` extra when depending on *mkdocstrings*:
```

File: docs/handlers/overview.md [hunk 4]
```diff
@@ -4,8 +4,160 @@ A handler is what makes it possible to collect and render documentation for a pa
...
+
+Note that the experimental handler does not yet offer the same features as the legacy one.
+If you are making extensive use of the current (legacy) Python handler selection and rendering options,
+you might want to wait a bit before trying the experimental handler.
+It is also not completely ready to handle dynamically built objects,
+like classes built with a call to `type(...)`.
+For most other cases, the experimental handler will work just fine.
+
+If you want to keep using the legacy handler as long as possible,
+you can already specify the `python-legacy` extra when depending on *mkdocstrings*:
```
Comment (pawamoy) [inline docs/handlers/overview.md:25]:
```suggestion
you can depend on `mkdocstrings-python-legacy` directly,
or specify the `python-legacy` extra when depending on *mkdocstrings*:
```

File: docs/handlers/overview.md (hunk 2 above)
Comment (pawamoy) [reply docs/handlers/overview.md:61]: I've changed the wording a bit to reflect that extras are not actually mandatory and that users can depend on the handlers directly.

**Pull request merged**